File: prova2Exercicios/maiorPilha.java
package prova2Exercicios;

public class maiorPilha {

}

class Lista{
    CelulaLista inicio;
    CelulaLista fim;


    public CelulaLista maiorPilha(){
        return maiorPilha(inicio);
    }

    public CelulaLista maiorPilha(CelulaLista i){
        CelulaLista maior = null;
        int m = 0;
        while (i != null) {
            CelulaPilha j = i.topo;
            int novoM = 0;
            while (j != null) {
                novoM ++;
                j = j.prox;
            }
            if(novoM > m){
                m = novoM;
                maior = i;
            }
            i = i.prox;
        }
        return maior;
    }
}

class CelulaLista{
    CelulaPilha topo;
    CelulaLista prox;
}

class CelulaPilha {
    int elemento;
    CelulaPilha prox;
}

File: prova2Exercicios/maiorPilha.c
#include <stdlib.h>

typedef struct CelulaPilha CelulaPilha;
typedef struct CelulaLista CelulaLista;

typedef struct CelulaPilha{
    int elemento;
    CelulaPilha *prox;
}CelulaPilha;

typedef struct CelulaLista{
    CelulaPilha *topo; 
    CelulaLista *prox;
} CelulaLista;

typedef struct Lista{
    CelulaLista *inicio;
    CelulaLista *fim;
} Lista;

CelulaLista *maiorPilha(CelulaLista *i){
    CelulaLista *maiorCelula = NULL;
    int maior = 0;
    while (i != NULL)
    {
        int atual = 0;
        CelulaPilha *j = i->topo;
        while (j != NULL)
        {
            atual++;
            j = j->prox;
        }
        if(atual > maior){
            maior = atual;
            maiorCelula = i;
        }
        i = i->prox;
        j = NULL;
    }
    i = NULL;
    return maiorCelula;
}


